drivers/hmis/app/models/hmis/ce/referral_message_handler.rb

```diff
@@ -93,6 +95,14 @@ def send_notification(message)
       # ).deliver_later
     end
 
+    def set_custom_referral_status(message) # rubocop:disable Naming/AccessorMethodName
+      status_key = message.params['custom_status_key']
+      return unless status_key
+
+      status = Hmis::Ce::CustomReferralStatus.find_by(key: status_key)
```

Comment:
suggest using the bang variant to raise if not found
```suggestion
      status = Hmis::Ce::CustomReferralStatus.find_by!(key: status_key, data_source: referral.data_source)
```